src/main/kotlin/no/nav/syfo/infrastructure/kafka/senoppfolging/SenOppfolgingSvarConsumer.kt

```diff
@@ -31,13 +30,24 @@ class SenOppfolgingSvarConsumer(private val senOppfolgingService: SenOppfolgingS
     }
 
     private fun processRecord(senOppfolgingSvarRecord: SenOppfolgingSvarRecord) {
-        // TODO: createKandidat should be moved to varsel consumer
-        val kandidat = senOppfolgingService.createKandidat(
+        val kandidatForVarsel = senOppfolgingSvarRecord.varselId?.let {
+            senOppfolgingService.findKandidatFromVarselId(it)
+        }
+        val recentKandidat = senOppfolgingService.findRecentKandidatFromPersonIdent(
             personident = Personident(senOppfolgingSvarRecord.personIdent),
-            varselAt = OffsetDateTime.now()
-        ).also { Metrics.COUNT_KAFKA_CONSUMER_SEN_OPPFOLGING_SVAR_KANDIDAT_CREATED.increment() }
+        )
+        val kandidat = if (kandidatForVarsel != null) {
+            kandidatForVarsel
+        } else if (recentKandidat != null) {
+            recentKandidat
+        } else {
+            senOppfolgingService.createKandidat(
+                personident = Personident(senOppfolgingSvarRecord.personIdent),
+            ).also {
+                Metrics.COUNT_KAFKA_CONSUMER_SEN_OPPFOLGING_SVAR_KANDIDAT_CREATED.increment()
+            }
+        }
 
-        // TODO: Should get kandidat based on varsel associated with svar, then add svar to that kandidat
         val onskerOppfolging = senOppfolgingSvarRecord.response.toOnskerOppfolging()
         senOppfolgingService.addSvar(
```

Comment:
Som vi snakket om i går så burde vi nok (foreløpig) kaste feil her dersom kandidaten allerede har svart. Når esyfo gjør det mulig å endre svar får vi tilpasse det (blir vel noe logikk avhengig av om kandidaten er ferdigbehandlet osv :thinking: )

Reply:
Fikset nå, men logger error i stedet for å kaste en exception (ellers vil jo meldingen aldri bli konsumert og alt stoppe opp).